CRM/Upgrade/Incremental/php/FiveThirtySeven.php

```diff
@@ -139,4 +140,45 @@ public static function updateDBDefaultsForContributionRecur(CRM_Queue_TaskContex
     return TRUE;
   }
 
+  /**
+   * Install recaptcha extension.
+   *
+   * This feature is restructured as a core extension - which will eventually allow us to replace/remove the
+   * reCAPTCHA implementation
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   * @throws \CRM_Core_Exception
+   */
+  public static function installReCaptchaExtension(CRM_Queue_TaskContext $ctx) {
+    // Install via direct SQL manipulation. Note that:
+    // (1) This extension has no activation logic.
+    // (2) On new installs, the extension is activated purely via default SQL INSERT.
+    // (3) Caches are flushed at the end of the upgrade.
+    // ($) Over long term, upgrade steps are more reliable in SQL. API/BAO sometimes don't work mid-upgrade.
+    $insert = CRM_Utils_SQL_Insert::into('civicrm_extension')->row([
+      'type' => 'module',
+      'full_name' => 'recaptcha',
+      'name' => 'reCAPTCHA',
+      'label' => 'reCAPTCHA',
+      'file' => 'recaptcha',
+      'schema_version' => NULL,
+      'is_active' => 1,
+    ]);
+    CRM_Core_DAO::executeQuery($insert->usingReplace()->toSQL());
+
+    CRM_Core_DAO::executeQuery('
+UPDATE civicrm_navigation
+SET name="Misc (Undelete, PDFs, Limits, Logging, etc.)"
+WHERE name="Misc (Undelete, PDFs, Limits, Logging, Captcha, etc.)"
```

Comment:
Generally speaking, I don't like the idea of changing machine names of things just because the labels change. That potentially breaks backward-compatibility and not for a very good reason (because no one but developers ever see them). But if we are going to change this name, can we please change it to something that's not __GODAWFUL__. I mean, WFT civicrm? What kind of machine name is that??

Reply:
Agree, it's a terrible name. Change it.